### PR TITLE
Created table of contents for the documentation.

### LINKED ISSUE #390: Create a table of content in doc/
Use doc/features.md for inspiration

### PR DESCRIPTION
**Pull request by ajeklund**
Markdown version of what will be generated by sphinx later on.

Closes #390.

### CODE REVIEW

File: doc/contents.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+This is a temporary table of contents, used during the development phase.
+When completed, the table of contents will be written in pydoc/index.rst
+and built with sphinx.
+
+Table of contents
+-----------------
+
+1. Introduction
+    * What is DLite and what is it used for?
+    * [Features](features.md)
```
Comment (jesper-friis) [inline doc/contents.md:13]: This be easier to read if we change it to [Features] and then add the following line at the end of the document:

[Features]: features.md

We want the links to point to the generated html-documentation, so the links at the end of the document must be updated when we know where the generated html files ends up on GitHub Pages.

Same comment for all other links

File: doc/contents.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+2. Installation instructions
+3. [Build instructions](build_with_vs.md)
```
Comment (jesper-friis) [inline doc/contents.md:15]: Here I suggest that we have a main build_instructions.md file with build instructions that we link to here.
The build_with_vs.md file can be referred to from build_instructions.md.

File: doc/contents.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+4. Tutorial
+5. Basic topics
+    * [Concepts](concepts.md)
+    * Data types
+    * [Mappings](mappings.md)
+    * [Collections](collections.md)
+    * Bindings
+6. [Tips and tricks](tips_and_tricks.md)
```
Comment (jesper-friis) [inline doc/contents.md:23]: The tips_and_tricks.md is currently for developers. 
I will suggest to add a "Developers guide" after "Advanced topics" under which we can have the documentation guideline and tips and tricks.

File: doc/contents.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+7. [Tools](tools.md)
+8. Advanced topics
+    * [Environment variables](environment_variables.md)
+    * [Storage plugins](storage_plugins.md)
+    * Mapping plugins
```
Comment (jesper-friis) [inline doc/contents.md:28]: Writing mapping plugins

File: doc/contents.md [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+    * [Code generation](code_generation.md)
+    * [Transactions](transactions.md)
```
Comment (jesper-friis) [inline doc/contents.md:30]: I think this could come at the end of Basic usage

**Review by jesper-friis — DISMISSED** [review]
Added some suggestions

File: doc/contents.md (hunk 1 above)
Comment (ajeklund) [reply doc/contents.md:13]: It seems like the line to be added got lost in the rendering of your comment (?).

The html-documentation will be generated by sphinx, which has a special syntax for the table of contents (which enables correct links to html once sphinx has been run). So, don't you think it makes sense to keep the links to the markdown files at this stage?

Comment (ajeklund) [reply doc/contents.md:13]: Keeping inline links to markdown files. This simplifies the creation of the sphinx version of the table of contents.

File: doc/contents.md (hunk 2 above)
Comment (ajeklund) [reply doc/contents.md:15]: Implemented as suggested.

File: doc/contents.md (hunk 3 above)
Comment (ajeklund) [reply doc/contents.md:23]: Implemented as suggested.

File: doc/contents.md [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    * [Concepts](concepts.md)
+    * Data types
+    * [Mappings](mappings.md)
+    * [Collections](collections.md)
+    * Bindings
+6. [Tips and tricks](tips_and_tricks.md)
+7. [Tools](tools.md)
+8. Advanced topics
+    * [Environment variables](environment_variables.md)
+    * [Storage plugins](storage_plugins.md)
```
[the remark this answers is not in the document]
Comment (ajeklund) [reply doc/contents.md:27]: Implemented as suggested.

File: doc/contents.md (hunk 4 above)
Comment (ajeklund) [reply doc/contents.md:28]: Implemented as suggested.

**Review by quaat — COMMENTED** [review]
On many sites, they are using sphinx-panels to separate sections on "Getting Started", "User Guide", "API reference" and "Developer Guide" 
examples:
  <https://pandas.pydata.org/docs/>
  <https://numpy.org/doc/stable/index.html>

would it be logical to include the Advanced Topics, such as writing plugins, into the developers guide?

File: doc/contents.md [hunk 7]
```diff
@@ -12,20 +12,22 @@ Table of contents
...
+7. Advanced topics
     * [Environment variables](environment_variables.md)
-    * [Storage plugins](storage_plugins.md)
-    * Mapping plugins
+    * [Writing storage plugins](storage_plugins.md)
+    * Writing mapping plugins
     * [Code generation](code_generation.md)
-    * [Transactions](transactions.md)
+8. Developer's guide
+    * [Documentation guideline]
```
Comment (jesper-friis) [inline doc/contents.md:31]:
```suggestion
    * [Documentation guideline]
    * [Documentation testing]
```

File: doc/contents.md [hunk 8]
```diff
@@ -12,20 +12,22 @@ Table of contents
...
+    * [Tips and tricks](tips_and_tricks.md)
```
Comment (jesper-friis) [inline doc/contents.md:32]:
```suggestion
    * [Tips and tricks](tips_and_tricks.md)
 
       
[Documentation guideline]: documentation_contributions.md
[Documentation testing]: documentation_testing.md
```

**Review by jesper-friis — COMMENTED** [review]
See suggestions. The documentation_testing.md file is added in PR #402.

File: doc/build_instructions.md [hunk 9]
```diff
@@ -0,0 +1,19 @@
+Build instructions
+==================
+
+Content
+-------
+  1. [General build instructions](#general-build-instructions)
+  2. [Build with Visual Studio](#build-with-visual-studio)
```
Comment (CasperWA) [inline doc/build_instructions.md:7]:
```suggestion

1. [General build instructions](#general-build-instructions)
2. [Build with Visual Studio](#build-with-visual-studio)
```

File: doc/build_instructions.md [hunk 10]
```diff
@@ -0,0 +1,19 @@
...
+
+
+General build instructions
+-------------
+xxx
+
```
Comment (CasperWA) [inline doc/build_instructions.md:13]:
```suggestion
General build instructions
-------------

xxx
```

File: doc/build_instructions.md [hunk 11]
```diff
@@ -0,0 +1,19 @@
...
+
+Build with Visual Studio
+-------------
+[Instructions for Visual Studio]
```
Comment (CasperWA) [inline doc/build_instructions.md:17]:
```suggestion

[Instructions for Visual Studio]
```

File: doc/contents.md [hunk 12]
```diff
@@ -0,0 +1,33 @@
+DLite documentation
+===================
+
+This is a temporary table of contents, used during the development phase.
+When completed, the table of contents will be written in pydoc/index.rst
+and built with sphinx.
```
Comment (CasperWA) [inline doc/contents.md:6]:
```suggestion
When completed, the table of contents will be written in pydoc/index.rst and built with Sphinx.
```

**Review by CasperWA — CHANGES_REQUESTED** [review]

File: doc/contents.md (hunk 7 above)
Comment (ajeklund) [reply doc/contents.md:31]: Added "Testing examples in the documentation".

File: doc/contents.md (hunk 8 above)
Comment (ajeklund) [reply doc/contents.md:32]: Keeping inline links in the markdown table of contents.

**Review by jesper-friis — COMMENTED** [review]
Fine with me. I line the suggestion by Thomas for the overall structure with a very simple front page and separate sub-pages for "Getting Started", "User Guide", "API reference" and "Developer Guide"

**Review by CasperWA — APPROVED** [review]

**Pull request merged**